### PR TITLE
feat: Enhance the handle of plan actions

### LINKED ISSUE #2344: Execute multiple plan actions at the same time.
**Describe the feature you'd like**
Currently, we are able to execute only one planned task at a time. But the answer to a query might contain a series of actions that need to be executed at different intervals. So, we need to implement this feature.

**Additional context**
This feature needs to be implemented after PR #2342 is merged.

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by atm1504**
Fixes #2344 #2336 

Changes: 
- [x]  Automatically determines plan action, by searching for ```plan_delay```
- [x] Handles multiple actions at the same time
- [x] Execute even when the app is in background
- [x] Implement stop actions.
- [x] Implement sound action for alarm.
- [x] Implement stop action for alarms
- [x] Look for possible errors.

Screenshots for the change: 
ezgif com-video-to-gif

Apk:

[app-fdroid-debug.apk.zip](https://github.com/fossasia/susi_android/files/3512699/app-fdroid-debug.apk.zip)

### CODE REVIEW

**Comment by atm1504** [comment]
@iamareebjamal please start reviewing it.

**Comment by atm1504** [comment]
@iamareebjamal please review it

File: app/src/main/java/org/fossasia/susi/ai/chat/ChatActivity.kt [hunk 1]
```diff
@@ -414,7 +414,7 @@ class ChatActivity : AppCompatActivity(), IChatView {
     }
 
     override fun databaseUpdated() {
-        rv_chat_feed.smoothScrollToPosition(recyclerAdapter.itemCount)
+        // rv_chat_feed.smoothScrollToPosition(recyclerAdapter.itemCount)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/ChatActivity.kt]: Why?

File: app/src/main/java/org/fossasia/susi/ai/chat/ChatPresenter.kt [hunk 2]
```diff
@@ -66,13 +69,10 @@ class ChatPresenter(context: Context) :
...
     var tableItem: TableItem? = null
-    private val planHandler: Handler = Handler()
-    lateinit var youtubeVid: IYoutubeVid
-    private var isPlanAction = false
-    private var planQueryAnswer: String = " "
+    private val youtubeVid: IYoutubeVid = YoutubeVid(context)
     private var textToSpeech: TextToSpeech? = null
-    private var isVideo = false
-    private lateinit var planVideoSusiResponse: SusiResponse
+    private val context: Context = context
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/ChatPresenter.kt:74]: Why?

Comment (atm1504) [reply app/src/main/java/org/fossasia/susi/ai/chat/ChatPresenter.kt:74]: I was failing to access the ```context``` from any functions. So, I declared it as a global variable, and this value has been used then.

File: app/src/main/java/org/fossasia/susi/ai/chat/ChatActivity.kt (hunk 1 above)
Comment (atm1504) [reply app/src/main/java/org/fossasia/susi/ai/chat/ChatActivity.kt]: This line was overwritten. The task, performed by this line is already been done when notifyDatabaseUpdated is called. Repeating the line was leading to the crash of the app, sometimes. So, I commented it out.

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/susi/ai/chat/ChatActivity.kt]: Remove it then. Commenting out was done in 1980s when there was no version control

Comment (atm1504) [reply app/src/main/java/org/fossasia/susi/ai/chat/ChatActivity.kt]: Updated it.

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**